Friendly-plans/app/src/main/java/pg/autyzm/friendly_plans/TaskContainerFragment.java

```diff
@@ -31,24 +34,86 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
 
     @Override
     public void onViewCreated(View view, Bundle savedInstanceState) {
+        registerViews(view);
+        taskNext.setOnClickListener(new View.OnClickListener() {
+                                        public void onClick(View v) {
+                                            if (isAllValidationOK()) {
+                                                createTaskInDb(); // fulfill the method to create task on db
+                                                goToNextPage();   // and start next activity
+                                            }
+                                        }
+                                    }
+        );
+
+    }
+
+
+    private boolean isAllValidationOK() {
```

Comment:
I would rather extract validation to separated class and not store the logic in fragment. Otherwise the fragment will grow up quickly. Let's add TaskValidator that will inherit Validator, and use it here. Further you could also inject it instead of making static class.